delay/delay.go

```diff
@@ -98,6 +104,43 @@ func isContext(t reflect.Type) bool {
 	return t == stdContextType || t == netContextType
 }
 
+// fileKey finds a stable representation of the caller's file path.
+// For calls from package main: strip all leading path entries, leaving just the filename.
+// For calls from anywhere else, strip $GOPATH/src, leaving just the package path and file path.
+func fileKey(file string) (string, error) {
+	if !internal.IsSecondGen() {
+		return file, nil
+	}
+	// If the caller is in the same Dir as mainPath, then strip everything but the file name.
+	if filepath.Dir(file) == appengine.MainPath {
+		return filepath.Base(file), nil
+	}
+	// If the path contains "_gopath/src/", which is what the builder uses for
+	// apps which don't use go modules, strip everything up to and including src.
+	// Or, if the path starts with /tmp/staging, then we're importing a package
+	// from the app's module (and we must be using go modules), and we have a
+	// path like /tmp/staging1234/srv/... so strip everything up to and
+	// including the first /srv/.
+	// And be sure to look at the GOPATH, for local development.
+	for _, s := range []string{"_gopath/src/", "/srv/", filepath.Join(build.Default.GOPATH, "src")} {
+		if idx := strings.Index(file, s); idx > 0 {
+			return file[idx+len(s):], nil
+		}
+	}
+
+	// Finally, if that all fails then we must be using go modules, and the file is a module,
+	// so the path looks like /go/pkg/mod/github.com/foo/bar@v0.0.0-20181026220418-f595d03440dc/baz.go
+	// So... remove everything up to and including mod, plus the @.... version string.
+	m := "/mod/"
+	if idx := strings.Index(file, m); idx > 0 {
+		file = file[idx+len(m):]
+	} else {
+		return file, fmt.Errorf("fileKey: unknown file path format for %q", file)
+	}
+	re := regexp.MustCompile("@v[^/]+")
```

Comment:
Better to move this to a package-level variable so it's evaluated once at package initialization:

```go
var modVersionPat = regexp.MustCompile(`@v[^/]+`)
```

Reply:
Done